### DISCUSSION TITLE
tbd4vcv rack beta mac

### DISCUSSION THREAD

**Discussion by ctag-fh-kiel** [ctag-fh-kiel/ctag-tbd#43] [Announcements]
tbd available for vcv rack --> <a href='https://github.com/ctag-fh-kiel/ctag-tbd/releases/tag/tbd4vcv-beta-mac'>tbd4vcv rack beta mac</a>

copy tbd4vcv-1.0.0-Darwin.zip file in "$RACK_DIR/plugins-v1" folder

once you have created an instance of tbd in rack you can change the parameters through a web server on localhost:3000 just like in the simulator. the active tbd instance linked to the web server can be selected through a context menu in vcv rack (ctrl-click on the module and activate web server).

comments in this discussion pls

**Comment by Motopomp** [comment]
After unzipping the archive is put the folder in my plugin folder and restart, TBD module, does not appear in the list of VCVRack modules.
OsX10.10.5 / VCVRack v1.1.6 
![Capture d’écran 2021-11-05 à 09 42 06](https://user-images.githubusercontent.com/33150728/140482653-d0dcbd7f-6111-4e6b-a5bb-7d96ba26208f.png)

**Reply by ctag-fh-kiel** [reply]
New version available
https://github.com/ctag-fh-kiel/ctag-tbd/releases/tag/tbd4vcv-beta-mac

**Comment by ctag-fh-kiel** [comment]
There were some issues with dynamic paths on VCV + some linking...
Can please try this one if it works now (pls erase old folder in plugins dir, then copy fresh .zip in plugins-v1). 
Thx :)

[https://github.com/ctag-fh-kiel/ctag-tbd/releases/tag/tbd4vcv-beta-mac](https://github.com/ctag-fh-kiel/ctag-tbd/releases/tag/tbd4vcv-beta-mac)

**Reply by Motopomp** [reply]
I'm sorry, I did not succeed, the situation does not change ... :(

**Reply by ctag-fh-kiel** [reply]
Strange, can you download it [here pls](https://drive.google.com/file/d/1R1sThWkd4gEIMjAOdoXU1jw0WOdTJk6g/view?usp=sharing)

**Reply by ctag-fh-kiel** [reply]
if it doesn't work, Rack creates a .log file in the plugin folder, pls share, maybe I can trace down the issue

**Reply by Motopomp** [reply]
I just tried again, and it still doesn't work, sorry ...
I made a little video (very low quality :)), so that you can see what I'm doing, and if I make a mistake.
I also give you the log.txt file, and indeed there is an error when loading the plugin.

https://user-images.githubusercontent.com/33150728/140638225-9a66f5d8-28d1-42ba-b41e-1ae967ceeb2b.mp4

Log.txt:
[log.txt](https://github.com/ctag-fh-kiel/ctag-tbd/files/7491947/log.txt)

**Comment by pgatt** [comment]
Hi, is there any chance you can build for VCV Rack v2? I'm mainly testing new modules for devs at the moment in that, happy to help here on 1.1.6 too though if necessary.

**Reply by ctag-fh-kiel** [reply]
Will have a look at v2 once the API has fully settled, thanks for offering :)

**Comment by ctag-fh-kiel** [comment]
Thanks a lot, it seems as if some symbols are missing. Probably something
with the standard c++ libs, maybe a version mismatch.
Which OS X version are you using?

tbd4vcv/plugin.dylib:
dlopen(/Users/azf-recording/Documents/Rack/plugins-v1/tbd4vcv/plugin.dylib,
6): Symbol not found: __ZdlPvSt11align_val_t
  Referenced from:
/Users/azf-recording/Documents/Rack/plugins-v1/tbd4vcv/plugin.dylib
  Expected in: /usr/lib/libc++.1.dylib
 in /Users/azf-recording/Documents/Rack/plugins-v1/tbd4vcv/plugin.dylib

**Reply by Motopomp** [reply]
II am using Osx10.10.5 ...
Is it possible to update my C ++ library? or maybe reinstall it, some way to make sure it is up to date, complete or uncorrupted and properly accessible?

**Reply by ctag-fh-kiel** [reply]
Good question, it is probably related to a change in the libc++ version on your system vs. mine.

**Comment by ctag-fh-kiel** [comment]


**Reply by Motopomp** [reply]
the link does not work correctly, and if I do "right click / download cicble", the archive is only 9 byte

**Reply by Motopomp** [reply]
is it the same as in "release"?

**Reply by ctag-fh-kiel** [reply]
Please download from the draft release, I update the binaries there.

**Comment by ctag-fh-kiel** [comment]
Can you try the one up in the release link, I updated the binaries there
too. Sorry for the confusion, still learning about GitHub capabilities :)

Frequence Morte ***@***.***> schrieb am So., 7. Nov. 2021,
17:22:

**Reply by Motopomp** [reply]
I just tested the latest release, and it doesn't work either.
I am sorry.
Am I the only one to test your module for mac? do other people use it with my osx version or others?

**Reply by ctag-fh-kiel** [reply]
There are three Mac OS testers so far including you, both others have newer OS X. I use 10.14.6 it works on my system without issues.

Can you resend the last .log file, just to check if anything has changed?
I am not sure if one can update libstdc++ on the Mac, as it seems you run an older version. Strange though that your other plugins work fine. tbd4vcv uses c++17.

**Reply by Motopomp** [reply]
the log file:
[log.txt](https://github.com/ctag-fh-kiel/ctag-tbd/files/7492992/log.txt)

**Comment by Visuelle-Musik** [comment]
Hi there - from your log.txt:
1.445 warn src/plugin.cpp:158] Could not load plugin /Users/azf-recording/Documents/Rack/plugins-v1/tbd4vcv: Failed to load library /Users/azf-recording/Documents/Rack/plugins-v1/tbd4vcv/plugin.dylib: dlopen(/Users/azf-recording/Documents/Rack/plugins-v1/tbd4vcv/plugin.dylib, 6): **Symbol not found: __ZdlPvSt11align_val_t**
  Referenced from: /Users/azf-recording/Documents/Rack/plugins-v1/tbd4vcv/plugin.dylib
  **Expected in: /usr/lib/libc++.1.dylib**
 in /Users/azf-recording/Documents/Rack/plugins-v1/tbd4vcv/plugin.dylib

I found two some cases of similar issues with  __ZdlPvSt11align_val_t missing in /libc++.1.dylib
https://forum.camunda.io/t/zbctl-status-doesnt-indicate-leader/374/6
https://github.com/yuzu-emu/yuzu/issues/1534

And they suggest, that **macOS 10.13 or higher** should solve the issue.
I have no idea if there is a way to supply the missing symbol/function in any other way on your system and a search on the net only suggested solutions for other missing functions by installing C-Environments like GCC and pointing that their runtime-libraries, like here:
https://stackoverflow.com/questions/50247343/dyld-symbol-not-found-zdapvm-running-king-on-mac-os-x
But afaik, the binary for TBD4VCV on MacOS was built using CLang, so I doubt that anything of that ilk would help?

**Comment by ctag-fh-kiel** [comment]
I have updated the release again, there was an issue with some paths. 
The c++17 issue with older OS X isn't fixed however, maybe someone with an older Mac could try to compile tbd4vcv from source?

**Reply by Motopomp** [reply]
Before I compiled myself, but for lack of space I had to get rid of some of my software and xcode is actually part with these almost 10go, I only compiled modules for VCV, to help, offer feedback to the developer , as with you.
Then the community grew and a lot of people (speaking better English than me) started to compile, so I stopped and got the place;).
So I am registered on two Facebook groups, the official and the French-speaking one, I started to prospect, in my language, and I would ask the question elsewhere according to the answers.
As a last resort, I would address myself directly to the site's forum, which is also very active, and that I advise you, if you are not already registered.
You could certainly find a lot more help there than I could ever give you, not being a develloper (I understand the advice, but I do not know any programming language)

**Reply by emptyvesselnz** [reply]
> I have updated the release again, there was an issue with some paths. The c++17 issue with older OS X isn't fixed however, maybe someone with an older Mac could try to compile tbd4vcv from source?

I've installed on my M1 Mac mini running OSX 11.6.1 and all looks fine, I do have an older Macbook still on 10.13 though so I can assist in testing and perhaps compiling if necessary.

**Reply by Motopomp** [reply]
Emptyvesselnz, we are searching for osx 10.10 compiler

**Comment by Motopomp** [comment]
Re. You must have seen that a Beta (1) version of VCVRack v2 was released, and they fixed, I don't know how, the compatibility issue with lower osx, like mine and below.
I test them, and it works ...
On the other hand, I have not found a person to compile your modules in the French-speaking group, so I asked the question today in the official group (English-speaking), I will remain the forum of the official site ...

**Comment by Moa13** [comment]
hi, are there any plans in the pipeline for MiRack @IOS release?

**Comment by ctag-fh-kiel** [comment]
Nope, not at this moment, but if you would like to take a stab at it, I am
happy to support...
